test/integration/smoke/test_vnf_templates.py

```diff
@@ -0,0 +1,338 @@
+
+        testClient = super(TestVnfTemplates, cls).getClsTestClient()
+        cls.apiclient = testClient.getApiClient()
+        cls._cleanup = []
+        cls.services = testClient.getParsedTestDataConfig()
+
+        # Get Zone, Domain and templates
+        cls.hypervisor = cls.testClient.getHypervisorInfo()
+        zone = get_zone(cls.apiclient, cls.testClient.getZoneForTests())
+        cls.zone = Zone(zone.__dict__)
+        cls.template = get_template(cls.apiclient, cls.zone.id)
+
+        cls.domain = Domain.create(
+            cls.apiclient,
+            cls.services["domain"]
+        )
+        cls.account = Account.create(
+            cls.apiclient,
+            cls.services["account"],
+            admin=True,
+            domainid=cls.domain.id
+        )
+        cls.user = cls.account.user[0]
+        cls.user_apiclient = cls.testClient.getUserApiClient(
+            cls.user.username, cls.domain.name
+        )
+
+        cls.service_offering = ServiceOffering.create(
+            cls.apiclient,
+            cls.services["service_offerings"]["big"]
+        )
+
+        cls._cleanup = [
+            cls.service_offering,
+            cls.domain,
+            cls.account
+        ]
+
+        cls.vnf_template_config = {
+            "name": "pfsense",
+            "displaytext": "pfsense",
+            "format": "QCOW2",
+            "url": cls.template.url,
+            "requireshvm": "True",
+            "ispublic": "True",
+            "isextractable": "True",
+            "hypervisor": cls.hypervisor,
+            "zoneid": cls.zone.id,
+            "ostype": "FreeBSD 12 (64-bit)",
+            "directdownload": False
+        }
+
+        cls.initial_setting = Configurations.list(
+            cls.apiclient,
+            name="vnf.template.appliance.enabled")[0].value
+
+        Configurations.update(cls.apiclient, "vnf.template.appliance.enabled", "true")
+
+        cls.vnf_templates = []
+
+    @classmethod
+    def tearDownClass(cls):
+        Configurations.update(cls.apiclient, "vnf.template.appliance.enabled", cls.initial_setting)
+        super(TestVnfTemplates, cls).tearDownClass()
+
+    def setUp(self):
+        self.apiclient = self.testClient.getApiClient()
+        self.dbclient = self.testClient.getDbConnection()
+        self.cleanup = []
+
+    def tearDown(self):
+        super(TestVnfTemplates, self).tearDown()
+
+    @attr(tags=["advanced"], required_hardware="false")
+    def test_01_register_vnf_template(self):
+        """Test register VNF template
+        """
+        self.vnf_template = VnfTemplate.register(self.user_apiclient,
+                                                 self.vnf_template_config,
+                                                 zoneid=self.zone.id,
+                                                 hypervisor=self.hypervisor,
+                                                 vnfnics=VNF_NICS,
+                                                 vnfdetails=VNF_DETAILS)
```

Comment:
should this be unregistered after the test?

Reply:
template will be removed in `test_05_delete_vnf_template`

Reply:
I think these (tests 1 and 5) should be part of setup and teardown. In this way the order of the tests matters, which can lead to issues of intermittend failures. Tests, even integration tests should be atomic.